apm-commons/apm-util/src/main/java/org/apache/skywalking/apm/util/ConfigInitializer.java

```diff
@@ -52,50 +52,50 @@ private static void initNextLevel(Properties properties, Class<?> recentConfigTy
             if (Modifier.isPublic(field.getModifiers()) && Modifier.isStatic(field.getModifiers())) {
                 String configKey = (parentDesc + "." + field.getName()).toLowerCase();
                 Class<?> type = field.getType();
-
                 if (Map.class.isAssignableFrom(type)) {
                     /*
                      * Map config format is, config_key[map_key]=map_value, such as plugin.opgroup.resttemplate.rule[abc]=/url/path
-                     * "config_key[]=" will generate a bleak Map , user could use this mechanism to set a blank Map
+                     * "config_key[]=" will generate an empty Map , user could use this mechanism to set an empty Map
                      */
                     ParameterizedType genericType = (ParameterizedType) field.getGenericType();
                     Type[] argumentTypes = genericType.getActualTypeArguments();
                     Type keyType = argumentTypes[0];
                     Type valueType = argumentTypes[1];
-                    // A chance to set a blank map
+                    // A chance to set an empty map
                     if (properties.containsKey(configKey + "[]")) {
-                        field.set(null, initBlankMap(type));
+                        Map currentValue = (Map) field.get(null);
+                        if (currentValue != null && !currentValue.isEmpty()) {
+                            field.set(null, initBlankMap(type));
```

Comment:
This method is still not changed.

Reply:
Sorry , I don't  understand .
If  the default value of Map isn't empty , I should  create  an empty Map .

Reply:
`initBlankMap`->`initEmptyMap`. A map can't be blank, a string can.

Reply:
Sorry .